### DISCUSSION TITLE
How to let Cypress For Jira automatically create Jira issues on failed test?

### DISCUSSION THREAD

**Discussion by AliOzeir** [cypress-io/cypress#22240] [Questions and Help]
I am planning to have a test automation workflow using Cypress, Jira, and Bitbucket pipelines where:

- I write my Cypress tests

- Integrate them into the bitbucket pipelines 

- Give the project key found in my Cypress Dashboard as an option in the bitbucket pipelines  

- Having my results in Cypress Dashboard after a successful run through the pipelines

- let the Jira issues automatically be raised on failure tests

I know that I can create a Jira issue manually after navigating to the Cypress dashboard but I wonder if there's a way to let Cypress For Jira raise them automatically. Thanks in advance

**Comment by samnittala** [comment]
You can leverage the JIRA APIs for this. 
Long back we built an integration just similar to the use case above using the JIRA APIs.

Atlassian's documentation is pretty good to get you started.

**Reply by madhuri155** [reply]
Hi can you please help me out how to update the jira issues using jira apis I didn't find any related article for this

**Comment by mudit94** [comment]
hi team, 
can you please share the sample code for this
